src/ucm/cuda/cudamem.h

```diff
@@ -20,10 +20,18 @@ CUresult ucm_cuMemAllocPitch(CUdeviceptr *dptr, size_t *pPitch,
 CUresult ucm_cuMemAllocPitch_v2(CUdeviceptr *dptr, size_t *pPitch,
                                 size_t WidthInBytes, size_t Height,
                                 unsigned int ElementSizeBytes);
+#if CUDA_VERSION >= 11020
```

Comment:
the `#if ` is probably not needed here since it's just func prototype, it's not calling any Cuda API here

Reply:
Done.

Reply:
The lack of `#if` fails for older CUDA because `cudaMemPool_t`/`CUmemoryPool` only appeared in CUDA 11.2. Seems like we need those directives anyway, or is there some better way of handling that case?

Reply:
For the time being I reinstated the `#if`s. I guess one alternative would be to define `cudaMemPool_t`/`CUmemoryPool` ourselves for older CUDA versions, but feels like that would pollute code more so not sure this is a great alternative.